pkg/net/net.go

```diff
@@ -50,6 +50,36 @@ func FormatAddress(addr string) string {
 	return addr
 }
 
+// AddressContainsPort checks to see if the supplied address contains both an address and a port.
+// This will not catch every possible permutation, but it is a best-effort routine suitable for prechecking human-interactive parameters.
+func AddressContainsPort(addr string) bool {
+	pieces := strings.Split(addr, ":")
```

Comment:
can we simplify it to:

* if `addr` doesn't contain ':', we let it through
* if `addr` contains `:`, but it parses as valid IP, we let it through?

Reply:
For the case we are looking at right now, that would be fine.  I was making the check a bit more elaborate to be useful in the other direction, too:  checking in cases where an address _should_ contain a port.

I'm not really sure; maybe it _is_ overly-elaborate for what it needs to be.

Reply:
Oh, sorry, no.  That still isn't sufficient:  the problem is that we need to be able to accept DNS _names_ as well as IPs.

Reply:
However, we _can_ branch early in the absence of any `:`, which seems like a reasonable simplification.

Reply:
yep, sounds like I thought about `AddressDoesntContainPort` which might be simpler